rfcs/20250618-pr-rules/README.md

```diff
@@ -0,0 +1,207 @@
+
+## Section 1: Opening a PR, Ready for Review
+
+1. PR Template and Task Checklist
+   * All PRs must use the provided template, including the task checklist.
+2. Draft PRs
+   * PRs should initially be opened as Draft.
+3. Marking PRs as Ready for Review
+   * Before marking a PR as Ready for Review:
+     * All tasks in the checklist must be completed.
+     * Features and changes must be fully implemented.
+     * Both private and public CI pipelines must pass, or clear justification for any
+       failing tests must be provided.
+4. Reverting a PR Back to Draft
+   * Once a PR is converted from Draft to Ready for Review, it should only be
+     reverted to Draft if:
+     * Significant changes or new feature requests arise during the review.
+     * The requested changes depend on future dependencies (e.g., waiting for a new
+       release) or require substantial time to implement.
+     * Other blockers prevent the changes from being merged into the current main branch.
+
+## Section 2: General Rules
+
+1. A PR cannot introduce new failures in GitHub checks (public CI) (i.e. stays
+green) or new failures in private CI (pre-commit\*).
+2. Removal of CI tests can only be allowed with the written and documented
+consent\*\* of the code owners in extreme circumstances.
+3. Code which causes new CI failures or problems should apply additional
+fixes within a day, otherwise the PR should be reverted. 
+4. Code scans often do not run with PRs, as such merged PRs should not increase
+the number of issues for the Intel internal code scans\*\*\*, and may be
+grounds for a reversion.
+5. A merged PR which causes a code scan failure should be fixed before the next
+scheduled code scan can be reverted.
+6. A designated person is assigned (for a determined period of time), who
+determines which PR was responsible for the failure.
+7. All PRs must have a label describing the purpose.
+8. Suggested labels are:
+   * bug (Section 3: Bug Fix)
+   * dependencies (Section 4: Dependencies)
+   * docs (Section 5: Documentation)
+   * enhancement (Section 6: Feature)
+   * infra (Section 7: Infrastructure)
+   * perf (Section 8: Performance)
+   * testing (Section 9: Tests)
+   * API/ABI breaking change (Section 10: API/ABI changes)
+9. Associated reviewers from CODEOWNERS must be included in review based on the
+   purpose/type of the PR, for example:
+   * Cases which require special consideration - CI failures/issues
+   * Changes to public facing header files etc.
+   * Changes to build structure (e.g. makefiles, bazel. etc.)
+11. Commits should follow commit message rules.
+12. During a code-freeze period (before drop), PRs should only be pulled in by
+main maintainers.
+13. An approval should be given with the expectation to merge, use best
+judgement in reviewing and approving PRs (get help as when not confident).
+14. The submitter is responsible for the maintenance (including closing) of
```

Comment:
i would put this broader as submitter is responsible for pushing PR towards integration